src/components/Maps/Sidebar.tsx

```diff
@@ -134,31 +243,46 @@ const Sidebar = ({
           {title}
         </Typography>
       </Collapse>
-      <ValidatorForm
-        onSubmit={(e) => {
-          onComplete({
-            label,
-            latitude,
-            longitude,
-            address,
-          });
-          setAddress('');
-          setLabel('');
-          e.preventDefault();
-        }}
-      >
+      <ValidatorForm onSubmit={(e) => onSubmit(e)}>
         <Collapse in={!isAutocompleteClicked}>
           <Typography variant="body1" classes={{ root: styles.label }}>
             Name:
           </Typography>
           <TextValidator
-            placeholder="Pin name"
+            placeholder={`${placeholderType} name`}
             onChange={handleLabelChange}
             value={label}
             validators={['required']}
-            errorMessages={['A pin name is required']}
+            errorMessages={[
+              `A ${placeholderType.toLowerCase()} name is required`,
+            ]}
             className={styles.pinLabelField}
           />
+          {(mode === MapModes.NewEvent || mode === MapModes.EditEvent) && (
+            <>
+              <Typography variant="body1" classes={{ root: styles.label }}>
+                Date:
+              </Typography>
+              <Container classes={{ root: styles.datePickerField }}>
+                <MuiPickersUtilsProvider utils={DateFnsUtils}>
+                  <ThemeProvider theme={datePickerTheme}>
+                    <KeyboardDatePicker
+                      disableToolbar
+                      variant="inline"
+                      format="MM/dd/yyyy"
+                      margin="normal"
+                      id="date-picker-inline"
+                      value={date}
+                      onChange={(date) => setDate(date)}
+                      KeyboardButtonProps={{
+                        'aria-label': 'change date',
+                      }}
+                    />
+                  </ThemeProvider>
+                </MuiPickersUtilsProvider>
+              </Container>
```

Comment:
this should be styled to match the name/location input fields that are also in the sidebar (size, font weight, etc.)